packages/ember-runtime/lib/computed/reduce_computed_macros.js

```diff
@@ -5,24 +5,101 @@
 
 import Ember from 'ember-metal/core'; // Ember.assert
 import { get } from 'ember-metal/property_get';
-import {
-  isArray,
-  guidFor
-} from 'ember-metal/utils';
+import { isArray } from 'ember-metal/utils';
 import EmberError from 'ember-metal/error';
 import {
-  forEach
+  forEach,
+  addObject,
+  indexOf,
+  intersection,
+  map as utilMap,
+  filter as utilFilter
 } from 'ember-metal/enumerable_utils';
-import run from 'ember-metal/run_loop';
-import { addObserver } from 'ember-metal/observer';
-import { arrayComputed } from 'ember-runtime/computed/array_computed';
-import { reduceComputed } from 'ember-runtime/computed/reduce_computed';
-import SubArray from 'ember-runtime/system/subarray';
-import keys from 'ember-metal/keys';
+import { ComputedProperty, computed } from "ember-metal/computed";
+import { addObserver, removeObserver } from "ember-metal/observer";
 import compare from 'ember-runtime/compare';
 
 var a_slice = [].slice;
 
+// TODO: Consider putting this with other array shims
```

Comment:
In 2.0.0 it seems like there might be an opportunity to take compat ES5 stuff into an 'ember.compat.js'.  This way developers can shed even more weight if they only support evergreen browsers.  It also draws a clean line for deprecations when ember decides not to support non ES5 browsers.